### PR TITLE
more form elements

### PR DESCRIPTION
**Pull request by grosser**
bringing checkbox positioning back to it's roots ...

before:
screen shot 2016-10-10 at 10 01 56 pm

after:

screen shot 2016-10-10 at 10 01 27 pm

@sandlerr 
/cc @zendesk/samson

### CODE REVIEW

File: app/views/stages/_fields.html.erb [hunk 1]
```diff
@@ -1,92 +1,34 @@
...
-          <%= form.label :production do %>
-            <%= form.check_box :production %>
-            Does this stage impact production servers?
-          <% end %>
-        </div>
-      </div>
-    </div>
+    <%= form.input :production, as: :check_box,
+          label: 'Impacts production',
+          help: "Deploys to production need to pass a buddy check when BuddyCheck is enabled."
```
Comment (sandlerr) [inline app/views/stages/_fields.html.erb]: should this message be conditional on `BuddyCheck.enabled?` ?

File: app/views/stages/_fields.html.erb [hunk 2]
```diff
@@ -126,16 +68,6 @@
...
-      Update Pull Request after deployment
-    <% end %>
-  </div>
-  <div class="col-lg-6 checkbox col-lg-offset-2">
-    <%= form.label :use_github_deployment_api do %>
-      <%= form.check_box :use_github_deployment_api %>
-      Use GitHub Deployment API [beta]
-    <% end %>
-  </div>
+  <%= form.input :update_github_pull_requests, as: :check_box, label: "Update Pull Request after deployment" %>
```
Comment (sandlerr) [inline app/views/stages/_fields.html.erb]: nitpick: Pull Requests (plural)

File: lib/samson/form_builder.rb [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
+        public_send(as, attribute, input_html)
       end
 
       content_tag :div, class: 'form-group' do
-        content = label(attribute, label, class: "col-lg-2 control-label")
-        content << content_tag(:div, class: 'col-lg-4', &block)
-        content << @template.additional_info(help) if help
-        content
+        if as == :check_box
+          content_tag(:div, class: "col-sm-offset-2 col-sm-10 checkbox") do
```
Comment (sandlerr) [inline lib/samson/form_builder.rb]: any idea why its `col-sm` here and `col-lg` below?

File: app/views/stages/_fields.html.erb (hunk 1 above)
Comment (grosser) [inline app/views/stages/_fields.html.erb]: ... I still want to allow people to set this even if they don't have buddy checks ... as documentation / for later use ... leaving it a bit ambiguous so I don't have to add extra logic :)

File: lib/samson/form_builder.rb [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
+            label(attribute) do
+              block.call.dup << SPACER << label << SPACER << help
```
Comment (sandlerr) [inline lib/samson/form_builder.rb]: why is it necessary to `dup`? If label and help are empty the spacers are being included for no reason... this is getting more complicated than I expected.

**Comment by grosser** [comment]
label will never be empty ... adding extra logic to help would also work ...
dup is needed because erb blocks return frozen strings ... need a test for
that :/

**Comment by grosser** [comment]
nvm ... the .dup was needed because I was passing a frozen string in the test ... updated that ...

**Comment by grosser** [comment]
... ah no ... inline erb block riggers it too ... keeping the tests frozen to reproduce it ... also less buggy to dup in case someone uses it with a block that returns a constant

**Comment by grosser** [comment]
also updated spacer logic ... either way seems fine to me and there is no visual difference ...

looks good ?

**Comment by grosser** [comment]
... so far loving the cleanliness for experienced users and that adding extra info is super easy for new users

**Pull request merged**